src/signpdf.js

```diff
@@ -99,7 +99,7 @@ export class SignPdf {
             bagType: forge.pki.oids.pkcs8ShroudedKeyBag,
         })[forge.pki.oids.pkcs8ShroudedKeyBag];
 
-        const privateKey = keyBags[0].key;
+        const privateKey = (keyBags[1] || keyBags[0]).key;
```

Comment:
Not quite convinced by this line. Actually matching should be done by key id here and not just guessing it's the first one.